src/core/src/op/if.cpp

```diff
@@ -33,15 +33,21 @@ static ov::PartialShape resolve_shape(const ov::PartialShape& then_pshape, const
         return ov::PartialShape::dynamic();
     }
     if (then_rank.get_length() != else_rank.get_length()) {
+        auto isScalar = [](const ov::PartialShape& pshape) {
+            return ((pshape.rank() == 0) || (pshape.rank() == 1 && pshape.get_shape()[0] == 1));
+        };
```

Comment:
```suggestion
        auto is_one_element = [](const ov::PartialShape& pshape) {
            return pshape.size() == 0 || pshape[0].get_max_length == 1;
        };
```
Should be sufficient as shape has static rank here.

Add test to validate new scenario, please.

Reply:
Done.

Reply:
Add/update test in [this file](https://github.com/openvinotoolkit/openvino/blob/5c55539e74efa2a4aab56e56c90a6f00e1dbeb49/src/core/tests/type_prop/if.cpp#L283)